account_payment_partner/models/account_payment_method.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountPaymentMethod(models.Model):
+    _inherit = 'account.payment.method'
```

Comment:
I would put this at payment mode level, not payment method. This way, using same SEPA direct debit payment method, you can define one payment mode showing full bank number and another not showing it for example.

Reply:
I don't agree.
I think that this has more relation with payment method.
In the low likely case that someone wants to show different things using the same payment method you can always duplicate it

Reply:
It's not only to show or not the bank account, but as you have added a very great feature of seeing last characters, first or full, it has to be at payment mode level for better allocation for that possibilities, or you will miss future updates on duplicated payment methods as they are maintained in `<data>` sections of the modules.